server/util/terminal/terminal_test.go

```diff
@@ -33,3 +36,192 @@ func TestTruncation(t *testing.T) {
 
 	// verify that we got here, no panic.
 }
+
+func TestComplexScreenWriting(t *testing.T) {
```

Comment:
Would it make sense/be possible to share the test cases between the two tests?

Reply:
Done!